src/components/Categories/CategoryFormDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import { Category } from "../../types";
+import { WithStyles, withStyles } from "@material-ui/core/styles";
+import { Theme, createStyles } from "@material-ui/core/styles";
+import EditIcon from "@material-ui/icons/Edit";
+import IconButton from "@material-ui/core/IconButton";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    button: { textTransform: "none" },
+    dialog: { width: 500 },
+    dialogTitle: {
+      backgroundColor: theme.palette.primary.dark,
+      color: theme.palette.common.white,
+    },
+  });
+
+interface Props {
```

Comment:
NIT better to name the props as a post fix to the component name like CategoryFormDialogProps

so if we export it later , then import it it will be import {CategoryFormDialogProps} from "path/to/CategoryFormDialog";